api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -1112,7 +1113,7 @@ message LoadBalancingPolicy {
   // Each client will iterate over the list in order and stop at the first policy that it
   // supports. This provides a mechanism for starting to use new LB policies that are not yet
   // supported by all clients.
-  repeated Policy policies = 1;
+  repeated core.v3.TypedExtensionConfig policies = 1;
```

Comment:
Reading the comment here, since xDS clients now report its capacities to control plane, I don't think we should make this `repeated` as a try list, selecting a right LB policy that supported by the client should be responsibility of the control plane, instead of sending a bunch of LB policies over the wire.

Reply:
SGTM. Tagging @markdroth, since you added this originally - are there any reasons to have `policies` be a repeated field beyond providing fallback behaviour for clients?

Reply:
We talked about this when I first added this field in https://github.com/envoyproxy/envoy/pull/7744#discussion_r316856468.  In general, I don't think we want to depend on client capabilities for this kind of thing, since that would put more burden on the control plane to have different code-paths for different clients.  I think it still makes sense for this to be an ordered list.

Reply:
I think that's the statement based on the fact that two years ago we don't have a sensible way to detect which client has what extensions built-in, now we report all extensions to control-plane so the basement of the argument changed.

Reply:
I don't think that having the client report all of its extensions to the server as part of its node information is actually all that useful, because the server can't modify the contents of resources based on that information without breaking the cacheability provided by [the new `xdstp:` URI-based naming scheme](https://github.com/cncf/xds/blob/main/proposals/TP1-xds-transport-next.md), since none of that information is part of the resource cache key.  I think that in order to make this work, we would need to find a way to either get the extension information into the context parameters, or else we'd need to figure out how to address this use-case via my [proposed design for dynamically generated cacheable resources](https://github.com/cncf/xds/pull/10).

I think it's far simpler to just push this decision to the client.

CC @htuch

Reply:
Yeah, for cacheability it seems ideal not to have the server specialize in this case. That might not always hold, but I think the list here and corresponding semantics should work.

Reply:
OK if we agree this should be a data plane responsibilities that sounds fair to me.